Lesson17/Lesson17-sinem/script.js

```diff
@@ -0,0 +1,834 @@
+     - Logs "Even" if the number is even
+     - Logs "Odd" if the number is odd
+*/
+console.log("Exp.10");
+
+function evenOrOdd(number) {
+  if (typeof number !== "number") {
+    console.log("Please try again.");
+    return;
+  }
+  if (number % 2 == 0) {
+    console.log("Even");
+  } else {
+    console.log("Odd");
+  }
+}
+
+evenOrOdd(6);
+evenOrOdd(7);
+
+/*
+11. URL Protocol Checker
+   - Define a function `checkProtocol(url)` that converts the URL to lowercase
+     and checks if it starts with "https" using .startsWith().
+   - Log "Secure connection" if true, otherwise "Unsecure connection".
+*/
+console.log("Exp.11");
+
+function checkProtocol(url) {
+  if (typeof url !== "string") {
+    console.log("Please try again.");
+    return;
+  }
+  if (url.startsWith("https")) {
+    console.log("Secure connection " + "'" + url.toLowerCase() + "'");
+  } else {
+    console.log("Unsecure connection.");
+  }
+}
+
+checkProtocol(
+  "https://tRANSLATE.GOOGLE.com/?sl=en&tl=tr&text=%22Unsecure%20connection%22&op=translate"
+);
+checkProtocol(
+  "/translate.google.com/?sl=en&tl=tr&text=%22Unsecure%20connection%22&op=translate"
+);
+checkProtocol(45);
+
+/*
+12. Switch: Day of the Week
+   - Define a function `getDayOfWeek(num)` that uses a switch statement:
+     1 -> "Monday"
+     2 -> "Tuesday"
+     ...
+     7 -> "Sunday"
+     - Log the matched day or "Invalid day" if out of range.
+*/
+console.log("Exp.12");
+
+function getDayOfWeek(num) {
+  switch (num) {
+    case 1:
+      console.log("Monday");
+      break;
+    case 2:
+      console.log("Tuesday");
+      break;
+    case 3:
+      console.log("Wednesday");
+      break;
+    case 4:
+      console.log("Thursday");
+      break;
+    case 5:
+      console.log("Friday");
+      break;
+    case 6:
+      console.log("Saturday");
+      break;
+    case 7:
+      console.log("Sunday");
+      break;
+    default:
+      console.log("Invalid day");
+      break;
+  }
+}
+
+getDayOfWeek(7);
+getDayOfWeek(6);
+getDayOfWeek(5);
+getDayOfWeek("hi");
+
+// value == value
+/// value and type === value and type
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+console.log("Exp.13");
+
+function repeatWord(word, times) {
+  if (typeof word !== "string" || typeof times !== "number") {
+    console.log("Please try again.");
+    return;
+  } else {
+    console.log(word.repeat(times));
+  }
+}
+
+repeatWord("sinem ", 3);
+repeatWord("sinem ", 6);
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+console.log("Exp.14");
+
+function censorWord(sentence, target) {
+  if (typeof sentence !== "string" || typeof target !== "string") {
+    console.log("Please try again.");
```

Comment:
Provide a better error message here. E.g. 'Please provide sentence and target in the string format.'